doc/libpmemobj.3.md

```diff
@@ -1637,26 +1632,81 @@ Using **TX_PARAM_MUTEX** or **TX_PARAM_RWLOCK** means that at the beginning of a
+**TX_PARAM_CB** registers a specified callback function to be executed at each transaction stage. For **TX_STAGE_WORK** it's executed before commit, for all other
 stages as a first operation after stage change. It will also be called after each transaction - in such case *stage* parameter will be set to **TX_STAGE_NONE**.
 pmemobj_tx_callback must be compatible with:
 
 ```void func(PMEMobjpool *pop, enum pobj_tx_stage stage, void *arg)```
 
 *pop* is a pool identifier used in **pmemobj_tx_begin**(), *stage* is a current transaction stage and *arg* is a second parameter of **TX_PARAM_CB**.
-Without considering transaction nesting this mechanism can be deemed as an alternative method for executing code between stages (instead of **TX_ONCOMMIT**,
+This mechanism can be deemed as an alternative method for executing code between stages (instead of **TX_ONCOMMIT**,
 **TX_ONABORT**, etc).
-However there are 2 significant differences when nested transactions are used:
 
-+  Registered function is executed only in the most outer transaction (even if registered in the inner one).
+Note that **TX_PARAM_CB** does not replace **TX_ONCOMMIT**/**TX_ONABORT**/etc. macros. They can be used together - a callback will be executed *before* **TX_ONCOMMIT**/**TX_ONABORT**/etc. section.
+
+If **pmemobj_tx_begin**() is called on the same memory pool within an open transaction, it starts a nested transaction on the pool. A new transaction may be started in this way only if the current stage of the open transaction is **TX_STAGE_NONE** or **TX_STAGE_WORK**.
+If successful, the transaction stage changes to **TX_STAGE_WORK** and the function returns zero. Otherwise, thes stage changes to **TX_STAGE_ONABORT** and an error number is returned.
+Transactions nested this way are flattened. Committing the inner transaction does not commit the outer transaction, however errors in the inner transaction are propagated up to the outermost level, resulting in the interruption of the entire transaction.
+
+There are 2 significant differences between nesting transactions and using stage callbacks:
+
++  Registered callback functions are executed only in the most outer transaction (even if registered in the inner one).
 
 +  There can be only one callback in the whole transaction (it can't be changed in the inner transaction).
 
-Note that **TX_PARAM_CB** does not replace **TX_ONCOMMIT**/**TX_ONABORT**/etc. macros. They can be used together - a callback will be executed *before*
-**TX_ONCOMMIT**/**TX_ONABORT**/etc. section.
+**TX_PARAM_CB** can be used when the code dealing with transaction stage changes is shared between multiple users or when it must be executed only in the outer transaction. For example it can be very useful when application must synchronize persistent and transient state.
+
+**pmemobj_tx_begin**() can also be called within an open transaction on a different memory pool. However, for this to succeed the current *transaction context* has to be changed.
```

Comment:
This has nothing to do with "different memory pools", but simply with changing the internal thread-local transaction. Please use as generic terms as possible, only specifying exact use cases as examples.